src/ssl.c

```diff
@@ -22184,8 +22184,10 @@ int wolfSSL_BN_hex2bn(WOLFSSL_BIGNUM** bn, const char* str)
 
         if (*bn == NULL)
             WOLFSSL_MSG("BN new failed");
-        else if (wolfSSL_BN_bin2bn(decoded, decSz, *bn) == NULL)
+        else if (wolfSSL_BN_bin2bn(decoded, decSz, *bn) == NULL) {
             WOLFSSL_MSG("Bad bin2bn error");
+            wolfSSL_BN_free(*bn); /* Free new BN */
```

Comment:
We should only free *bn if we created *bn a few lines above.  If the user passed in a valid *bn then we should NOT free it.  I think we need to distinguish those cases here.

Reply:
Good catch Todd! I missed that case. We should only free it if created above at line 22195. You could do something similar to line 21942 with weOwn.

Reply:
Oh right! Updated with Todd's suggestion.